internal/testutil/acc/config_formatter_test.go

```diff
@@ -104,3 +108,257 @@ func TestFormatToHCLLifecycleIgnore(t *testing.T) {
+    num_shards = 1
+    zone_name  = "Zone 2"
+
+    region_configs {
+      priority      = 7
+      provider_name = "AWS"
+      region_name   = "EU_WEST_2"
+      auto_scaling {
+        disk_gb_enabled = false
+      }
+      electable_specs {
+        instance_size = "M10"
+        node_count    = 3
+      }
+    }
+  }
+
+}
+`
+var twoRegionConfigs = `
+resource "mongodbatlas_advanced_cluster" "cluster_info" {
+  backup_enabled = false
+  cluster_type   = "REPLICASET"
+  name           = "my-name"
+  project_id     = "project"
+
+  replication_specs {
+    num_shards = 1
+    zone_name  = "Zone 1"
+
+    region_configs {
+      priority      = 7
+      provider_name = "AWS"
+      region_name   = "US_WEST_1"
+      auto_scaling {
+        disk_gb_enabled = false
+      }
+      electable_specs {
+        instance_size = "M10"
+        node_count    = 3
+      }
+    }
+
+    region_configs {
+      priority      = 7
+      provider_name = "AWS"
+      region_name   = "EU_WEST_1"
+      auto_scaling {
+        disk_gb_enabled = false
+      }
+      electable_specs {
+        instance_size = "M10"
+        node_count    = 3
+      }
+    }
+  }
+
+}
+`
+
+func Test_ClusterResourceHcl(t *testing.T) {
+	var (
+		clusterName                 = "my-name"
+		replicationSpecMultiRegions = admin.ReplicationSpec{
+			NumShards: conversion.IntPtr(1),
+			ZoneName:  conversion.StringPtr("Zone 1"),
+			RegionConfigs: &[]admin.CloudRegionConfig{
+				acc.CloudRegionConfig("US_WEST_1", constant.AWS, "M10", 3),
+				acc.CloudRegionConfig("EU_WEST_1", constant.AWS, "M10", 3),
+			},
+		}
+		testCases = map[string]struct {
+			expected string
+			req      acc.ClusterRequest
+		}{
+			"defaults": {
+				standardClusterResource,
+				acc.ClusterRequest{ClusterNameExplicit: clusterName},
+			},
+			"dependsOn": {
+				dependsOnClusterResource,
+				acc.ClusterRequest{ClusterNameExplicit: clusterName, ResourceDependencyName: "mongodbatlas_project.project_execution"},
+			},
+			"dependsOnMulti": {
+				dependsOnMultiResource,
+				acc.ClusterRequest{ClusterNameExplicit: clusterName, ResourceDependencyName: fmt.Sprintf("%s, %s", "mongodbatlas_private_endpoint_regional_mode.atlasrm", "mongodbatlas_privatelink_endpoint_service.atlasple")},
+			},
+			"twoReplicationSpecs": {
+				twoReplicationSpecs,
+				acc.ClusterRequest{ClusterNameExplicit: clusterName, ReplicationSpecs: []admin.ReplicationSpec{
+					acc.ReplicationSpec(&acc.ReplicationSpecRequest{Region: "US_WEST_1", ZoneName: "Zone 1"}),
+					acc.ReplicationSpec(&acc.ReplicationSpecRequest{Region: "EU_WEST_2", ZoneName: "Zone 2"}),
+				}},
+			},
+			"overrideClusterResource": {
+				overrideClusterResource,
+				acc.ClusterRequest{ClusterNameExplicit: clusterName, Geosharded: true, CloudBackup: true, ReplicationSpecs: []admin.ReplicationSpec{
+					acc.ReplicationSpec(&acc.ReplicationSpecRequest{
+						Region: "MY_REGION_1", ZoneName: "Zone X", InstanceSize: "M30", NodeCount: 30, ProviderName: "AZURE",
```

Comment:
knit: values are disconnected from the string, e.g. overrideClusterResource in this case.   We might want to use constants fro "M30", 30, etc, and use them in both the TF string and rep spec params

Reply:
So far we haven't use any constants for Region or InstanceSize. Uncertain if it adds that much value

Reply:
it's that i can that it's easier to follow why instance type is M30, and you have to go to overrideClusterResource.

but it's ok to leave it as it is.